extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AutoAddScopeProcessor.java

```diff
@@ -67,12 +85,17 @@ public void transform(TransformationContext context) {
                         }
                     }
                     if (autoScope.test(clazz, context.getAnnotations(), beanArchiveIndex.getIndex())) {
-                        context.transform().add(autoScope.getDefaultScope()).done();
+                        if (scope != null) {
+                            LOGGER.debugf("Scope %s was already added for reason: %s", scope, reason);
```

Comment:
I originally had WARNING here but TBH I'm not sure if it really helps...

Reply:
Maybe we should rather add the possibility to handle this kind of situation (e.g. throw an exception) inside the build item?

Reply:
I think we shouldn't worry about it for now. I don't think there is a high chance of it becoming a problem

Reply:
There's always a chance... I have it ready anyway so I'll update the PR ;-)

Reply:
okay :)

Reply:
> Maybe we should rather add the possibility to handle this kind of situation (e.g. throw an exception) inside the build item?

+1, I think we want to give the user a chance to react to this in some way.
Exception is one of them, forceful override of the scope could be another (i.e. replacement of scope).
Or, in case we don't want to hold user responsible for configuration setup, we should treat it as an erroneous situation and throw an exception ourselves as soon as this transformer gets invoked (so that it's visible where it's coming from). WDYT?

Reply:
My original idea of the contract was: higher priority wins but other ``AutoAddScopeBuildItem``s can log a warning or throw an exception. 

And if a scope was added in another transformer before ``AutoAddScopeBuildItem``s are processed then just log a warning (note that all ``AutoAddScopeBuildItem``s are processed in one annotation transformer.)

Reply:
> My original idea of the contract was: higher priority wins but other AutoAddScopeBuildItems can log a warning or throw an exception.

Yea, that would be good since having multiple scopes is always a problem.

>  (note that all AutoAddScopeBuildItems are processed in one annotation transformer.)

Ah, thanks for the reminder, I always seem to forget that :)